### PR TITLE
Update imagepullsecret and update to v0.13.0

### LINKED ISSUE #54: Add support for set_image_pull_secrets
Check whether I can use this as an example:
[set_image_pull_secrets](https://github.com/kubeflow/pipelines/blob/264ff37c1e21b413ec1ccc5049541429dc2d9fd5/sdk/python/kfp/dsl/_pipeline.py#L66)

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #54 

**Description of your changes:**
- Fix taskRunSpecs bugs for v0.13.0 release (there's a missing `s` in the spec)
- Update imagepullsecret to use the Tekton v0.13.0 podTemplate (also help for the single PipelineRun PR)
- Update default Tekton version to v.0.13.0 since we will be moving forward with these new features.

**Environment tested:**

* Python Version (use `python --version`): 3.6.4
* Tekton Version (use `tkn version`): v0.13.0
* Kubernetes Version (use `kubectl version`): 1.16
* OS (e.g. from `/etc/os-release`):

### CODE REVIEW

File: sdk/FEATURES.md [hunk 1]
```diff
@@ -82,7 +82,16 @@ The [affinity](/sdk/python/tests/compiler/testdata/affinity.py),
 [node_selector](/sdk/python/tests/compiler/testdata/node_selector.py), and
 [tolerations](/sdk/python/tests/compiler/testdata/tolerations.py) python tests are examples of how to use these features.
 
-This feature is recently implemented in Tekton and is available on the Tekton master branch and Tekton 0.13.0+.
+This feature is recently implemented in Tekton and is available on Tekton v0.13.0 and above.
+
+### ImagePullSecrets
+
+ImagePullSecret is a feature for the component to know which secret to use when pulling container images from private registries. It is implemented
```
Comment (animeshsingh) [inline sdk/FEATURES.md:89]:
```suggestion
ImagePullSecret is a feature for the components to know which secret to use when pulling container images from private registries. It is implemented
```

File: sdk/FEATURES.md [hunk 2]
```diff
@@ -82,7 +82,16 @@ The [affinity](/sdk/python/tests/compiler/testdata/affinity.py),
...
+with Tekton's [podTemplate](https://github.com/tektoncd/pipeline/blob/master/docs/podtemplates.md) field under Tekton
+PipelineRun. The [imagepullsecrets](/sdk/python/tests/compiler/testdata/imagepullsecrets.py) python test is an example of how to use this
+feature.
+
+This feature is recently implemented in Tekton and is available on Tekton v0.13.0 and above.
```
Comment (animeshsingh) [inline sdk/FEATURES.md:94]:
```suggestion
This feature is recently implemented in Tekton and is available on Tekton v0.13.0 onwards.
```

File: sdk/FEATURES.md [hunk 3]
```diff
@@ -197,6 +195,5 @@ An exit handler is a component that always executes, irrespective of success or
 
 [ParallelFor]: https://github.com/tektoncd/pipeline/issues/2050
 [VarSub]: https://github.com/tektoncd/pipeline/issues/2322
```
Comment (fenglixa) [inline sdk/FEATURES.md:197]: VarSub was also addressed in KFP-tekton, is it?

**Comment by ckadner** [comment]
@Tomcli -- can you address Feng's and Animesh's comments so we can merge this? Thanks.

File: sdk/FEATURES.md (hunk 2 above)
Comment (Tomcli) [reply sdk/FEATURES.md]: Done

File: sdk/FEATURES.md (hunk 1 above)
Comment (Tomcli) [reply sdk/FEATURES.md]: Done

File: sdk/FEATURES.md (hunk 3 above)
Comment (Tomcli) [reply sdk/FEATURES.md]: VarSub is not resolved, the issue is being redirected to a different one with the same feature requirement. I updated the issue link to the new issue.

**Pull request merged**